File: LeetCode/0771-jewels-and-stones/0771-jewels-and-stones.java
class Solution {
    public int numJewelsInStones(String jewels, String stones) {
        int answer = 0;
        Map<Character, Integer> jewelsMap = new HashMap<>();
        
        for (char jewel : jewels.toCharArray()) {
            jewelsMap.put(jewel, jewelsMap.getOrDefault(jewel, 0) + 1);
        }
        
        for (char stone : stones.toCharArray()) {
            if (jewelsMap.containsKey(stone)) {
                answer++;
            }
        }
        
        return answer;
    }
}

File: LeetCode/0771-jewels-and-stones/0771-jewels-and-stones.kt
class Solution {
    fun numJewelsInStones(jewels: String, stones: String): Int {
        var answer: Int = 0
        val freqs: MutableSet<Char> = HashSet()
        
        jewels.forEach{ jewel -> freqs.add(jewel) }
        stones.forEach{ stone ->
            if (freqs.contains(stone)) {
                answer++
            }
        }
        
        return answer
    }
}